### PR TITLE
Chat fe newest

### PR DESCRIPTION
**Pull request by hskowron**
Need to finish up putting the rest of the socket functions in place

### CODE REVIEW

File: ui/src/views/Profile/ProfileView.tsx [hunk 1]
```diff
@@ -62,6 +68,17 @@ export default function ProfileView(): React.ReactElement {
     mode: "onChange",
   });
 
+  React.useEffect(() => {
```
Comment (williamollio) [inline ui/src/views/Profile/ProfileView.tsx:71]: What do you want to do here ? I think this won't be stable. Let's have a talk about that

**Comment by williamollio** [comment]
This files inside the hooks folder in the ui part might need to be put in the services folder. And might be wrapped inside a class or several ones. What do you think ?

**Comment by williamollio** [comment]
In the axiosInstance, the base URL of the BE is already defined

File: ui/src/views/Profile/ProfileView.tsx (hunk 1 above)
Comment (hskowron) [reply ui/src/views/Profile/ProfileView.tsx:71]: It is to set the status to Online, we will have to do that for every url. In what way will this not be stable?

File: nest/src/channel/channel.gateway.ts [hunk 2]
```diff
@@ -31,15 +34,35 @@ enum acknoledgementStatus {
 @WebSocketGateway(3333, {
   cors: {
     credentials: true,
+    origin: 'http://localhost:3000',
```
Comment (moabid42) [inline nest/src/channel/channel.gateway.ts:37]: That's good to strict the origin to a specific urls, but it's better to define them in the env file! smth like this :
```
    origin: [
      process.env.FRONTEND_URL,
      process.env.BACKEND_URL,
    ],
```

File: nest/src/users/users.controller.ts [hunk 3]
```diff
@@ -40,7 +40,7 @@ import {
 } from './utils/upload-utils';
 
 @Controller('users')
-@UseGuards(JwtGuard)
+// @UseGuards(JwtGuard)
```
Comment (moabid42) [inline nest/src/users/users.controller.ts:43]: why did you remove the guard from here ?

File: nest/src/users/users.gateway.ts [hunk 4]
```diff
@@ -12,12 +12,18 @@ import { GetCurrentUserId } from '../decorators/getCurrentUserId.decorator';
 import { socketToUserId } from './socketToUserIdStorage.service';
 import { UsersService } from './users.service';
 import * as msgpack from 'socket.io-msgpack-parser';
+import { JwtGuard } from 'src/auth/guards/jwt.guard';
 
 // add some cors sanitazation here
 @WebSocketGateway(8888, {
+  cors: {
+    credentials: true,
+    origin: 'http://localhost:3000',
```
Comment (moabid42) [inline nest/src/users/users.gateway.ts:21]: same here add this too to the env variable!

File: nest/src/channel/channel.service.ts [hunk 5]
```diff
@@ -420,9 +426,9 @@ export class ChannelService {
       }
       /* Hash the password */
       if (typeof dto.passwordHash === 'string' && dto.type === 'PROTECTED') {
-        dto.passwordHash = await argon.hash(dto.passwordHash, {
-          type: argon.argon2id,
-        });
+        dto.passwordHash = dto.passwordHash; // await argon.hash(dto.passwordHash, {
```
Comment (moabid42) [inline nest/src/channel/channel.service.ts:429]: why did you remove the hashing here again ?

**Review by moabid42 — COMMENTED** [review]
Nice logic for the channel invite! I will try to update the currentPasswordHash later!

File: nest/src/users/users.controller.ts (hunk 3 above)
Comment (williamollio) [reply nest/src/users/users.controller.ts:43]: yeah that's probably to run some requests on swagger when the user isn't authenticated yet

File: ui/src/views/Profile/ProfileView.tsx (hunk 1 above)
Comment (williamollio) [reply ui/src/views/Profile/ProfileView.tsx:71]: I am actually not sure anymore, we can run some tests together later on

File: nest/src/channel/channel.service.ts (hunk 5 above)
Comment (hskowron) [reply nest/src/channel/channel.service.ts:429]: I am having issues with argon, it just crashes the backend. I is definetely an issue on my end but I am not sure how to fix it, so to work I have to comment out all the lines that use argon.

File: nest/src/channel/channel.gateway.ts (hunk 2 above)
Comment (moabid42) [reply nest/src/channel/channel.gateway.ts:37]: Yeah it is always good to hide the origins !

File: nest/src/channel/channel.service.ts (hunk 5 above)
Comment (moabid42) [reply nest/src/channel/channel.service.ts:429]: Fair enough maybe @williamollio can take a look if he have time or we can take a look together on thursday

Comment (williamollio) [reply nest/src/channel/channel.service.ts:429]: Sure

File: nest/src/channel/channel.gateway.ts (hunk 2 above)
Comment (hskowron) [reply nest/src/channel/channel.gateway.ts]: So it seems there is currently a bug with react-scripts version 5, where process is not defined. @williamollio do you know anything about that by chance?

**Review by williamollio — APPROVED** [review]
You did a very good job !
